docs/navigation-configuration.md

```diff
@@ -81,6 +81,40 @@ navigation: {
 ...
 ```
 
+### Best practices for navigation structuring
+
+<!-- add-attribute:class:warning -->
+>**NOTE:** Do not use `getConfig` and `setConfig` methods to extend `navigation.nodes` configuration. Unwanted side effects might occur.
+
+We encourage you to use (async) functions to build the node tree, since there are limitations if you try to update the `navigation.nodes` with `setConfig` and `configChanged('navigation')`. Each `node.children` which is of type function, gets executed every time it's parent node is routed to.
```

Comment:
Can they only use async functions or both async and non-async? I'm not sure what you meant with the parenthesis. 

```suggestion
We encourage you to use functions or [async functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) to build the node tree, since there are limitations if you try to update the `navigation.nodes` with `setConfig` and `configChanged('navigation')`. Each `node.children` which is of type function gets executed every time its parent node is routed to.
```